src/astra_driver.cpp

```diff
@@ -62,8 +62,15 @@ AstraDriver::AstraDriver(ros::NodeHandle& n, ros::NodeHandle& pnh) :
     color_subscribers_(false),
     depth_subscribers_(false),
     depth_raw_subscribers_(false),
-    uvc_flip_(0)
+    uvc_flip_(0),
+    min_diagnostics_freq_(8.0),
+    max_diagnostics_freq_(std::numeric_limits<double>::quiet_NaN()),
+    freq_stat_color_(diagnostic_updater::FrequencyStatusParam(&min_diagnostics_freq_, &max_diagnostics_freq_, 0.0, 10)),
+    freq_stat_depth_(diagnostic_updater::FrequencyStatusParam(&min_diagnostics_freq_, &max_diagnostics_freq_, 0.0, 10))
 {
+  pnh_.param<double>("min_diagnostics_freq", min_diagnostics_freq_, min_diagnostics_freq_);
+  pnh.setParam("min_diagnostics_freq", min_diagnostics_freq_);
```

Comment:
Why are you setting the parameter here?

Reply:
The parameter is needed to have an option to set it dynamically at the runtime, so it should be also set with the initial value on start.